client/director.go

```diff
@@ -213,105 +137,93 @@ func getCachesFromDirectorResponse(resp *http.Response, needsToken bool) (caches
 
 		// Construct the cache objects, getting endpoint and auth requirements from
 		// Director
-		var cache namespaces.DirectorCache
-		cache.AuthedReq = needsToken
-		cache.EndpointUrl = endpoint
-		cache.Priority = pri
-		caches = append(caches, cache)
+		server, err := url.Parse(endpoint)
+		if err != nil {
+			log.Errorln("Failed to parse server:", endpoint, "error:", err)
+			continue
+		}
+		serversPrio = append(serversPrio, ServerPriority{URL: server, Priority: pri})
 	}
 
 	// Making the assumption that the Link header doesn't already provide the caches
 	// in order (even though it probably does). This sorts the caches and ensures
 	// we're using the "pri" tag to order them
-	sort.Slice(caches, func(i, j int) bool {
-		val1 := caches[i].Priority
-		val2 := caches[j].Priority
-		return val1 < val2
+	sort.Slice(serversPrio, func(i, j int) bool {
+		return serversPrio[i].Priority < serversPrio[j].Priority
 	})
 
-	return caches, err
+	servers = make([]*url.URL, len(serversPrio))
+	for i, serverPrio := range serversPrio {
+		servers[i] = serverPrio.URL
+	}
+
+	return
 }
 
-// NewTransferDetails creates the TransferDetails struct with the given cache
-func newTransferDetailsUsingDirector(cache namespaces.DirectorCache, opts transferDetailsOptions) []transferAttemptDetails {
-	details := make([]transferAttemptDetails, 0)
-	cacheEndpoint := cache.EndpointUrl
+// Retrieve federation namespace information for a given URL.
+func GetDirectorInfoForPath(ctx context.Context, resourcePath, directorUrl string, isPut bool, query string) (parsedResponse server_structs.DirectorResponse, err error) {
+	if directorUrl == "" {
+		return server_structs.DirectorResponse{},
+			errors.Errorf("unable to retrieve information from a Director for object %s because no director URL was provided", resourcePath)
+	}
 
-	// Form the URL
-	cacheURL, err := url.Parse(cacheEndpoint)
```

Comment:
Quite a few of these errors deal with falling back to a "dirlisthots" for the given namespace -- a concept that comes from Topology and isn't something we'd get from any modern Director. And while the Director previously supplied a `dirlisthost` in its `X-Pelican-Namespace` header, this hasn't been the case for many, many months. After 4 or 5 major releases, I think we're safe dropping the backwards compatibility for this edge case, especially because we keep our central services up to date and I don't think there's anyone out there running something this old.

Reply:
I think you answered a different question here, the one about the errors, this was a question about no longer keeping the DirectorCache.

Reply:
Whoops, you're right.

This `DirectorCache` isn't actually any kind of TTL information cache -- it's an interface that allowed us long ago to handle constructing a "cache" object for transfers that needed a priority, along with an Auth/NonAuth URL (which aren't supplied by the Director). With the removal of topology, it's no longer the client's job to figure any of this stuff out, because it should be relying on the Director's enumeration of object server URLs/ports capable of providing the object.